### ISSUE TITLE
Feature Request: LaunchKey (Mini) support

### ISSUE THREAD

**Issue by Phoenix616** [ruohki/lunchpad#10]
**Is your feature request related to a problem? Please describe.**
I would love to see the ability to also use the buttons and maybe even the dials on a LaunchKey (Mini).

**Describe the solution you'd like**
The LaunchKey to be supported

**Describe alternatives you've considered**
Some other program or midi translator but I couldn't find a good solution that integrates with all the features of the LaunchKey.

**Additional context**
The goal here is mainly to be able to use the LaunchKey to control OBS too and, besides the normal button usage, also do more advanced stuff like using the dials or sliders for volume control or even manual fades.

**Comment by ruohki** [comment]
This is indeed planned. I just need one of these devices to grab all the midi codes and tinker with it. Or i need someone who can supply me with the needed midi message codes

**Comment by Phoenix616** [comment]
If you can point me into the right direction regarding how to easily get those message codes in the format you would need them then I could see what I can do about that.

I also did some testing with [Houston](https://obsproject.com/forum/resources/houston-midi-controller-for-obs.1046/) (which unfortunately has some issues with macros and OBS integration for me and is sadly not open source :eyes:) and managed to get some of the buttons setup (mainly the touch buttons) but some buttons might be hardcoded/not exposed as midi keys and the keyboard and touch pad seem to use the same notes for some. :S

If you can do anything with it, [here](https://github.com/ruohki/lunchpad/files/7500138/LaunchKeyMini.txt) is the Houston setup file that I did for the LaunchKey Mini MK3 when testing. (Top row is the dials which Houston only detects as single buttons, center the 2x8 touch key rows)

**Comment by ruohki** [comment]
Yes i can do that its mainly sysex specific code maybe there is an api document as for the launchpads. Mind sharing me a link to the device so i can take a look?

**Comment by Phoenix616** [comment]
[This](https://novationmusic.com/en/keys/launchkey-mini) is the one I'm talking about. Did some quick googling and found [this document](https://www.kraftmusic.com/media/ownersmanual/Novation_Launchkey_Programmers_Reference_Manual.pdf) which seems to include information necessary for that. (EDIT: Actually that's for the full one, not Mini but I assume the Mini has it's linked somewhere on the official site too but I couldn't find that with a quick search)

**Comment by ruohki** [comment]
Ill look into it - it has a lot of cool features like the sliders and knob, wonder if can use them - iam rewriting the project at the moment, but ill get one of the devices and try to support the basics (keys) as of 0.5 release

**Comment by Phoenix616** [comment]
I see, glad to hear you are at least considering this. Also I did some more searching and found the Programmers Reference Guide for the Launchkey MK3 [here](https://customer.novationmusic.com/en/support/downloads?brand=Novation&product_by_type=1441&download_type=technical-docs). Looks like it is really only linked on a full Launchkey for some reason, I guess we either have to do some more digging or I just do some testing once you have something to test :)

I also found [this gist](https://gist.github.com/llimllib/13a391bc52aea9af3c8e81a315647240) which seems to use the API and include some references to the codes as well as a [reverse engineered repo](https://github.com/giezu/LaunchkeyMiniMK3) for the Mini MK3 which might help (but it's obviously up to you if you want to look at such legally-questionable code...)

**Comment by ruohki** [comment]
If this is still needed, are we talking about this device:

<img width="679" height="356" alt="Image" src="https://github.com/user-attachments/assets/c9431775-505d-45bd-9da4-bfa4ff9e0b12" />

**Comment by Phoenix616** [comment]
> If this is still needed, are we talking about this device:
> 
> <img width="679" height="356" alt="Image" src="https://github.com/user-attachments/assets/c9431775-505d-45bd-9da4-bfa4ff9e0b12" />

Yes (to both)

**Comment by ruohki** [comment]
Found that model locally, gonna make it happen next week

**Comment by ruohki** [comment]
<img width="2795" height="1219" alt="Image" src="https://github.com/user-attachments/assets/00bbb8aa-dd61-4dbf-b18c-590d42e7d438" />

I will receive a Launchkey Mini Mk3 / 4 this week to test things on the real device. If things work well, support will be in rc.3

**Comment by Phoenix616** [comment]
Really cool! I can test things too if you want :)

**Comment by ruohki** [comment]
Just waiting for my device to arrive. Will verify on my end and then release the version with MK3 and MK4 support.

**Comment by ruohki** [comment]
Support is in rc3 wich is releasing right now
